src/hmmdwrkr.c

```diff
@@ -191,6 +211,280 @@ worker_process(ESL_GETOPTS *go)
+   * one-time config info from options
+   */
+  wrk = esl_gencode_WorkstateCreate(query->opts, gcode);
+
+  //if (query->cmd_type == HMMD_CMD_SEARCH) threadObj = esl_threads_Create(&search_thread);
+  //else
+    threadObj = esl_threads_Create(&scan_thread);
+
+
+  if (query->query_type == HMMD_SEQUENCE) {
+    fprintf(stdout, "Search seq %s  [L=%ld]", query->seq->name, (long) query->seq->n);
+  } else {
+    fprintf(stdout, "Search hmm %s  [M=%d]", query->hmm->name, query->hmm->M);
+  }
+  fprintf(stdout, " vs %s DB %d [%d - %d]",
+          (query->cmd_type == HMMD_CMD_SEARCH) ? "SEQ" : "HMM", 
+          query->dbx, query->inx, query->inx + query->cnt - 1);
+
+
+  if (range_list)
+     fprintf(stdout, " in range(s) %s", esl_opt_GetString(query->opts, "--seqdb_ranges"));
+
+  fprintf(stdout, "\n");
+
+
+  /* copy and convert the DNA sequence to text so we can print it in the domain alignment display */
+  qsqDNATxt = esl_sq_Create();
+  esl_sq_Copy(query->seq, qsqDNATxt);
+
+  printf("Creating 6 frame translations\n");
+  /* create sequence block to hold translated ORFs */
+  wrk->orf_block = esl_sq_CreateDigitalBlock(1024, abc);
+
+  /* translate DNA sequence to 6 frame ORFs */
+  do_sq_by_sequences(gcode, wrk, query->seq);
+
+  /* Create processing pipeline and hit list accumulators */
+  tophits_accumulator  = p7_tophits_Create(); 
+
+/*  if (query->cmd_type == HMMD_CMD_SEARCH) {
+      pipelinehits_accumulator = p7_pipeline_Create(query->opts, 100, 100, FALSE, p7_SEARCH_SEQS);
+  }
+  else {
+*/
+      pipelinehits_accumulator = p7_pipeline_Create(query->opts, 100, 100, FALSE, p7_SCAN_MODELS);
+//  }
+
+  /* process each 6 frame translated sequence */
+  for (k = 0; k < wrk->orf_block->count; ++k)
+   {
+     qsq = &(wrk->orf_block->list[k]);
+
+	 /* 
+	   use the name, accession, and description from the DNA sequence and
+	   not from the ORF which is generated by gencode and only for internal use
+	 */
+     if ((status = esl_sq_SetName(qsq, query->seq->name))   != eslOK)  p7_Fail("Set query sequence name failed in hmmdwrkr.c");
+     if ((status = esl_sq_SetAccession(qsq, query->seq->acc))    != eslOK)  p7_Fail("Set query sequence accession failed in hmmdwrkr.c");
+     if ((status = esl_sq_SetDesc(qsq, query->seq->desc))   != eslOK)  p7_Fail("Set query sequence description failed in hmmdwrkr.c");
+
+	 
+     /* Create processing pipeline and hit list */   
+     for (i = 0; i < env->ncpus; ++i) {
+        info[i].abc   = abc;
+        info[i].hmm   = query->hmm;
+        info[i].seq   = qsq;
+        info[i].opts  = query->opts;
+
+        info[i].range_list  = range_list;
+           
+        info[i].th    = NULL;
+        info[i].pli   = NULL;
+
+        info[i].ntseq = qsqDNATxt; /* for printing the DNA target sequence in the domain hits display */
+	
+        info[i].inx_mutex = &inx_mutex;
+        info[i].inx       = &current_index;/* this is confusing trickery - to share a single variable across all threads */
```

Comment:
Example of confusing copy pasta. This code is cut/pasted from elsewhere, complete with a TODO issue noted in comments. Now it's not clear that there's actually more than one related TODO in the code because of the copy-paste. Better to note something like "this came from XXX, see detailed comments over there, including a note about how this needs to be cleaned up both there and here".